### ISSUE TITLE
seastar - Failed to allocate 122880 bytes

### ISSUE THREAD

**Issue by bentsi** [scylladb/scylla#4735]
*Installation details*
Scylla version (or git commit hash): 3.1.0.rc2-0.20190713.92bf92817
Cluster size: 5
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-03d6caf668ad59911

nemesis in parallel:

- RestartThenRepairNode
- nodetool cleanup
- nodetool repair

db log (full log [ip-172-30-0-250.gz](https://github.com/scylladb/scylla/files/3402011/ip-172-30-0-250.gz))
```
2019-07-17T00:29:01+00:00  ip-172-30-0-250 !INFO    | scylla.bin: [shard 0] rpc - client 172.30.0.144:7001: fail to connect: Connection refused
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !ERR     | scylla.bin: [shard 0] seastar - Failed to allocate 122880 bytes
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: Aborting on shard 0.
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: Backtrace:
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000004164e92
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000004052115
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000004052415
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x00000000040524c3
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x00007f09fa31b02f
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: /opt/scylladb/bin/../libreloc/libc.so.6+0x000000000003853e
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: /opt/scylladb/bin/../libreloc/libc.so.6+0x0000000000022894
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x00000000040061ca
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000004007e14
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x000000000407e92a
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000004106c8a
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000004108144
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x000000000410a5e7
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x000000000404f601
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x000000000404f7fe
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x00000000041314cd
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000003fb99d6
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x0000000003fbb122
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x00000000007d6244
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: /opt/scylladb/bin/../libreloc/libc.so.6+0x0000000000024412
2019-07-17T04:00:25+00:00  ip-172-30-0-250 !INFO    | scylla: 0x000000000083654d
2019-07-17T04:02:17+00:00  ip-172-30-0-250 !INFO    | scylla-jmx: Exception in thread "Storage Service Repair" java.lang.IllegalStateException: Unable to connect to Scy
lla API server: java.net.SocketException: Connection reset
```

Translated backtrace:
```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::backtrace_buffer::append_backtrace() at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) print_with_backtrace at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:1104
seastar::print_with_backtrace(char const*) at /usr/include/boost/program_options/variables_map.hpp:146
sigabrt_action at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:5007
 (inlined by) _FUN at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:5003
?? ??:0
?? ??:0
?? ??:0
seastar::memory::on_allocation_failure(unsigned long) at memory.cc:?
__libc_posix_memalign at ??:?
 (inlined by) posix_memalign at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/memory.cc:1601
seastar::temporary_buffer<unsigned char>::aligned(unsigned long, unsigned long) at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) seastar::posix_file_impl::read_maybe_eof(unsigned long, unsigned long, seastar::io_priority_class const&) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:2402
seastar::posix_file_impl::dma_read_bulk(unsigned long, unsigned long, seastar::io_priority_class const&)::{lambda(unsigned long)#1}::operator()(unsigned long)::{lambda()#2}::operator()() at /usr/include/boost/program_options/variables_map.hpp:146
apply<seastar::posix_file_impl::dma_read_bulk(uint64_t, size_t, const seastar::io_priority_class&)::<lambda(size_t)> mutable::<lambda()>&> at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) apply<seastar::posix_file_impl::dma_read_bulk(uint64_t, size_t, const seastar::io_priority_class&)::<lambda(size_t)> mutable::<lambda()>&> at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/future.hh:1385
 (inlined by) do_until<seastar::posix_file_impl::dma_read_bulk(uint64_t, size_t, const seastar::io_priority_class&)::<lambda(size_t)> mutable::<lambda()>, seastar::posix_file_impl::dma_read_bulk(uint64_t, size_t, const seastar::io_priority_class&)::<lambda(size_t)> mutable::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/future-util.hh:507
 (inlined by) operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:2370
 (inlined by) apply at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/apply.hh:35
 (inlined by) apply<seastar::posix_file_impl::dma_read_bulk(uint64_t, size_t, const seastar::io_priority_class&)::<lambda(size_t)>, long unsigned int> at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/apply.hh:43
 (inlined by) apply<seastar::posix_file_impl::dma_read_bulk(uint64_t, size_t, const seastar::io_priority_class&)::<lambda(size_t)>, long unsigned int> at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/future.hh:1392
operator()<seastar::future_state<long unsigned int> > at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) run_and_dispose at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/future.hh:377
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::reactor::run_some_tasks() at /usr/include/boost/program_options/variables_map.hpp:146
seastar::reactor::run_some_tasks() at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) seastar::reactor::run() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:4238
seastar::app_template::run_deprecated(int, char**, std::function<void ()>&&) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/future.hh:768
seastar::app_template::run(int, char**, std::function<seastar::future<int> ()>&&) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/core/future.hh:768
main at crtstuff.c:?
?? ??:0
_start at ??:?
```

Coredump info:
```
backtrace=           PID: 62166 (scylla)
           UID: 996 (scylla)
           GID: 1001 (scylla)
        Signal: 6 (ABRT)
     Timestamp: Wed 2019-07-17 04:00:25 UTC (2min 2s ago)
  Command Line: /usr/bin/scylla /opt/scylladb/bin/../libexec/scylla.bin --blocked-reactor-notify-ms 500 --abort-on-lsa-bad-alloc 1 --abort-on-seastar-bad-alloc --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --network-stack posix --io-properties-file=/etc/scylla.d/io_properties.yaml --cpuset 1-7,9-15
    Executable: /opt/scylladb/libreloc/ld.so
 Control Group: /system.slice/scylla-server.service
          Unit: scylla-server.service
         Slice: system.slice
       Boot ID: c03a5087ce3546869ea8e80131c2a252
    Machine ID: df877a200226bc47d06f26dae0736ec9
      Hostname: ip-172-30-0-250.ec2.internal
      Coredump: /var/lib/systemd/coredump/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000
       Message: Process 62166 (scylla) of user 996 dumped core.
                
                Stack trace of thread 62166:
                #0  0x00007f09f9c2053f raise (/opt/scylladb/libreloc/libc.so.6)
                #1  0x00007f09f9c0a95e abort (/opt/scylladb/libreloc/libc.so.6)
                #2  0x00000000040061cb on_allocation_failure (/opt/scylladb/libexec/scylla.bin)
```

Coredump download instructions:
```
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.aa
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.ab
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.ac
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.ad
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.ae
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.af
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.ag
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.ah
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.ai
wget https://storage.cloud.google.com/upload.scylladb.com/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz/core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.aj

# To join coredump pieces, you may use:
cat core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz.* > core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz
# To decompress you may use:
unpigz --fast core.scylla.996.c03a5087ce3546869ea8e80131c2a252.62166.1563336025000000.gz
```

**Comment by slivne** [comment]
@bentsi on 3.4 does this happen

row level repair was updated in 3.2 / 3.3 to be better with regards to memory consumption - so if its related to repair it's probably fixed

**Comment by slivne** [comment]
I am closing this issue since repair was updated with regards to resource management if this happens in versions > 4.0 please open a new issue
